mesop/examples/docs/on_load.py

```diff
@@ -0,0 +1,16 @@
+import mesop as me
+
+
+def on_load(e: me.LoadEvent):
+  me.state(State).default_values = ["a", "b"]
```

Comment:
I feel like for an example, we want a better example here. In this case it's static, so technically one could use `field(lambda: ["a", "b"])` in the state class instead of the load event.

But I guess in the more dynamic case, one could also use: ``field(lambda: random.shuffle(["a", "b"])` as well.

I think that's the one thing about the `on_load`. I'm wondering what's a really good use case for it. I feel like there is one though. 

I mean using the `field()` function is bit weird since we import from dataclasses. But it does work.

Reply:
Good point, I updated this simulate a fake streaming API that's slow.

This is the main reason I added on_load because you'll often want to do some kind slow API call (e.g. RPC, open a file, etc), and the current trick of loading state inside a component state has no way of letting you yield intermediate states (esp. loading/in-progress states).